drivers: hwinfo: microchip: Add hardware info driver

Farsin-Nasar-Microchip opened the pull request:
- Added hwinfo node in the samd5xe5x.dtsi file.
- Added device tree binding file for Microchip HWINFO G1.
- Add new hardware info driver for Microchip G1 series.
- Update Kconfig and CMakeLists.txt to support the new driver.
- Update sam_e54_xpro.yaml to reflect G1 hwinfo support on the board.

Farsin-Nasar-Microchip committed:
dts: arm: microchip: samd5xe5x: Add HWINFO node and binding

- Added hwinfo node in the samd5xe5x.dtsi file.
- Added device tree binding file for Microchip HWINFO G1.

Signed-off-by: Farsin Nasar V A <farsin.nasarva@microchip.com>

Farsin-Nasar-Microchip committed:
drivers: hwinfo: microchip: Add hardware info driver

- Add new hardware info driver for Microchip G1 series.
- Update Kconfig and CMakeLists.txt to support the new driver.

Signed-off-by: Farsin Nasar V A <farsin.nasarva@microchip.com>

Farsin-Nasar-Microchip committed:
boards: microchip: sam_e54_xpro: add hwinfo to supported list

- Update sam_e54_xpro.yaml to reflect G1 hwinfo support on the board.

Signed-off-by: Farsin Nasar V A <farsin.nasarva@microchip.com>

nandojve commented on drivers/hwinfo/hwinfo_mchp_g1.c:
```diff
@@ -0,0 +1,97 @@
+	*cause = result;
+
+	return 0;
+}
```
```diff
}
+#endif /* DT_NODE_HAS_STATUS_OKAY(DT_NODELABEL(rstc)) */
```

alexanderwachter commented on boards/microchip/sam/sam_e54_xpro/sam_e54_xpro.yaml:
```diff
@@ -16,6 +16,7 @@ supported:
   - entropy
   - flash
   - gpio
+  - hwinfo
```
Since this API does not cost anything if it is not used (compiler will throw it away), I would just unconditionally always compile it. Adding it to every board is a bit exhausting.

alexanderwachter replied:
@Farsin-Nasar-Microchip we synced face to face. @nandojve will comment.

nandojve committed:
dts: arm: microchip: samd5xe5x: Fix rstc node

The hwinfo driver should build inconditionally. It checks if the rstc
node is present and when CONFIG_HWINFO=y is selected by the user the
rstc information will be available. The fallback mechanism will provide
a weak implementation for all APIs that are not available.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve reviewed (approved):
Thank you for the tips @alexanderwachter
